audio-worklet/noisegate-test.html

```diff
@@ -0,0 +1,451 @@
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<!DOCTYPE html>
+  <head>
+    <script src="test/testharness.js"></script>
+    <script src="test/audit-util.js"></script>
+    <script src="test/audit.js"></script>
+    <script src="src/noisegate-script-processor.js"></script>
+
+    <script id="noisegate-test-code">
+      // Sample rate is a power of two to avoid interpolation from delay node.
+      // It is held artificially low for easy inspection of audio buffers.
+      // All tests pass for all legal powers in the range [3000, 384000].
+      const sampleRate = Math.pow(2, 12);
+
+      const testDurationSec = 1;
+      const testFrames = testDurationSec * sampleRate;
+      const bufferSize = 512;
+      const timeConstant = 0.0025;
+      const scriptProcessorLatency = bufferSize * 2;
+
+      let audit = Audit.createTaskRunner();
+      
+      // With a threshold of linearThreshold and a constant source of some value
+      // above linearThreshold, followed by a sequence of values beneath
+      // linearThreshold, an ideal envelope follower would make the noise gate
+      // output equal to the input. But a proper envelope can only be
+      // approximated. This test verifies that the envelope delay is
+      // mathematically correct given a timeConstant.
+      audit.define('envelope-is-mathematically-correct', (task, should) => {
+        let context = new OfflineAudioContext(1, testFrames, sampleRate);
+        
+        // A timeToThreshold ten times larger than the timeConstant should get
+        // the envelope close enough to 1 (it never reaches 1) before the
+        // constant source node is scheduled to drop to 0.
+        const timeToThreshold = 10 * timeConstant;
+        
+        // The noise gate scales values for sine waves, which reach an
+        // an average power of a^2 / 2.
+        const linearThreshold = 0.5;
+        const scaledThreshold = 2 * linearThreshold;
+
+        // The noise gate bases muting decisions on a decibel scale.
+        const threshold = NoiseGate.toDecibel(scaledThreshold);
+
+        // Attack and release are both 0.
+        let noisegate =
+            new NoiseGate(context, {bufferSize, threshold, timeConstant});
+
+        // The signal should be all 1's prior to timeToThreshold to simplify
+        // the math that relates the closing index to the timeConstant.
+        let gain = new GainNode(context, {gain : 1});
+        let constantSource = new ConstantSourceNode(context, {offset: 1});
```

Comment:
Be consistent: Line 64 has "gain : 1", but line 65 has "offset: 1".  I think the preferred style is "gain: 1".